### PR TITLE
Grid sorting/filtering, events and ARIA attributes

### LINKED ISSUE #497: igx-grid cells aria-describedby is pointing to an id that doesn't exist
### Description  
The `igx-grid` `th` elements don't have ids.

### Steps to reproduce  

1. Open https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid.html
2. Inspect any grid cell and its `aria-describedby` attribute
3. Inspect the `th` the grid cell belongs to

### Result  
There is no id on the `th` element that corresponds to the `aria-describedby` attribute value.

### Expected result  
The `th` element id should be the same as the `aria-describedby` value.

### Attachments  
image

### PR DESCRIPTION
**Pull request by rkaraivanov**
- Added relevant ARIA attributes.
- Additional style classes exposed.
- Events are mirrored to the old API as close as possible.
- Removed the setter for the cell value. Updates should be done through the new exposed `update()` method.
- Row object now returns a reference to its native element (if available).
- "Streamlined" the API for filtering/sorting.

Also closes #497 #496

### CODE REVIEW

File: src/grid/api.service.ts [hunk 1]
```diff
@@ -86,8 +88,25 @@ export class IgxGridAPIService {
         this.get(id).filteringExpressions = filteringState;
     }
 
+    public filter_multiple(id: string, expressions: IFilteringExpression[]) {
```
Comment (kdinev) [inline src/grid/api.service.ts]: Why are there methods not camel-cased?

File: src/grid/grid-filtering.spec.ts [hunk 2]
```diff
@@ -253,86 +254,86 @@ describe("IgxGrid - Filtering actions", () => {
...
-        grid.clearFilter("ReleaseDate");
+        grid.clearFiltering("ReleaseDate");
         fix.detectChanges();
         grid.filter("ReleaseDate", null, DATE_FILTERS.notNull);
         fix.detectChanges();
         expect(grid.rowList.length).toEqual(6);
 
         // Empty filter
-        grid.clearFilter("ReleaseDate");
+        grid.clearFiltering("ReleaseDate");
```
Comment (kdinev) [inline src/grid/grid-filtering.spec.ts]: Is this a breaking change in the API?

File: src/grid/grid.component.ts [hunk 3]
```diff
@@ -301,41 +302,47 @@ export class IgxGridComponent implements OnInit, AfterContentInit {
...
     public filterGlobal(value: any, condition?, ignoreCase?) {
-        // TODO: AND OR Filtering logic
         this.gridAPI.filterGlobal(this.id, value, condition, ignoreCase);
-        this.page = 0;
     }
 
-    public clearFilter(name: string) {
-        const col = this.gridAPI.get_column_by_name(this.id, name);
-        if (!col) {
+    public clearFiltering(name?: string) {
```
Comment (kdinev) [inline src/grid/grid.component.ts]: Don't introduce such method naming changes. Revert this back to `clearFilter`

File: src/grid/grid.component.ts [hunk 4]
```diff
@@ -301,41 +302,47 @@ export class IgxGridComponent implements OnInit, AfterContentInit {
...
+        if (!this.gridAPI.get_column_by_name(this.id, name)) {
             return;
         }
-
         this.gridAPI.clear_filter(this.id, name);
     }
 
-    public clearFilterAll() {
-
+    public clearSorting(name?: string) {
```
Comment (kdinev) [inline src/grid/grid.component.ts]: To follow the `clearFilter` change this to `clearSort`

File: src/grid/cell.component.ts [hunk 5]
```diff
@@ -192,6 +189,13 @@ export class IgxGridCellComponent {
         this.handleInlineEditMode(event);
     }
 
+    public update(val: any) {
```
Comment (kdinev) [inline src/grid/cell.component.ts]: I guess changing a `value` setter to a modifier method introduces a breaking change. If this affects any of the documentation, readmes, etc. please pull updates to the related documentation. Check the getting-started page as well, for any code that may stop working.

File: src/grid/api.service.ts (hunk 1 above)
Comment (rkaraivanov) [reply src/grid/api.service.ts]: Those are internal from the point of the grid and are not exposed directly to the user.

Comment (kdinev) [reply src/grid/api.service.ts]: Then why is `filterGlobal` camel-cased?

**Pull request merged**